.github/workflows/build-release.yaml

```diff
@@ -0,0 +1,96 @@
+name: Build and release
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    tags:
+      - "**"
+
+concurrency:
+  group: ${{ github.repository_id }}-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: "MacOS x86"
+            runner: macos-12-large
+            release-suffix: macosx-amd64
+          - name: "MacOS arm64"
+            runner: [self-hosted, macOS, ARM64]
+            release-suffix: macosx-arm64
+          - name: "Linux x86"
+            runner: matterlabs-ci-runner
+            image: matterlabs/llvm_runner:ubuntu22-llvm17-latest
+            release-suffix: linux-amd64
+          - name: "Linux ARM64"
+            runner: matterlabs-ci-runner-arm
+            image: matterlabs/llvm_runner:ubuntu22-llvm17-latest
+            release-suffix: linux-arm64
+          - name: "Windows"
```

Comment:
Are we building < 0.6.0 binaries for Windows? The original build_and_release_binary didn't.

Reply:
Indeed, @antonbaliasnikov we do not support <0.6.0 on Windows. Devops said it was too problematic.

Reply:
@abinavpp though probably Anton wants to take it into account in each particular branch.
Let's wait for his response.

Reply:
> Devops said it was too problematic.

I completely reworked Windows build, no crazy powershell anymore, we do mingw64 and clang as for zksolc/zkvyper. So, in theory, there should be no difference, but I'll test it properly today and check if there are issues.

Reply:
After lots of testing and investigating the old builds, I made a decision to return the ugly powershell, because it's impossible to make mingw64 work with C++ code which is 5 years old and build system barely working :( 

Check for 0.6.0 added back.